### PR TITLE
Changed MD5 code for licencing reasons

### LINKED ISSUE #49: License Problem
There is a problem with the license because it cannot be released under the Apache 2 license and still contain GPL code.

### PR DESCRIPTION
**Pull request by inverse**
Resolves #49 

Would be good if you guys could test this before merging it in to ensure that it's all functional. Checked it here and it appeared to work fine though and scrobbled fine :+1: 

Built using: https://developer.android.com/reference/java/security/MessageDigest.html

### CODE REVIEW

File: app/src/main/java/com/adam/aslfms/util/MD5.java [hunk 1]
```diff
@@ -1,742 +1,54 @@
...
-
-	private static byte[] encode(int input[], int len){
-		byte[] out = new byte[len];
-		int i, j;
-		for (i = j = 0; j  < len; i++, j += 4) {
-			out[j] = (byte) (input[i] & 0xff);
-			out[j + 1] = (byte) ((input[i] >>> 8) & 0xff);
-			out[j + 2] = (byte) ((input[i] >>> 16) & 0xff);
-			out[j + 3] = (byte) ((input[i] >>> 24) & 0xff);
+		} catch (NoSuchAlgorithmException e) {
```
Comment (tgwizard) [inline app/src/main/java/com/adam/aslfms/util/MD5.java]: I'd suggest just rethrowing this as a runtime exception. Unless there's something sensible we can do if MD5 is not available? Is it ever not available?

**Comment by tgwizard** [comment]
Nice!

**Comment by inverse** [comment]
Should probably update the readme when this is merged too.

**Comment by inverse** [comment]
Any reason why this isn't getting merged? Want me to amend the licence in this PR or create another one post merge?

**Comment by HumbleBeeBumbleBee** [comment]
Sorry @inverse. Will be working on it tonight and tomorrow.

**Comment by HumbleBeeBumbleBee** [comment]
Tested on emulator with android 2.2 and 6.0
I believe that should be sufficient.

**Pull request merged**